sdk/iot/core/tests/cmocka/test_az_iot_core.c

```diff
@@ -157,13 +92,65 @@ static void _az_iot_u32toa_size_success(void** state)
   assert_int_equal(_az_iot_u32toa_size(4294967295), 10);
 }
 
+static void test_az_iot_is_success_status_translate_success()
+{
+  assert_true(az_iot_is_success_status(AZ_IOT_STATUS_OK));
+  assert_true(az_iot_is_success_status(AZ_IOT_STATUS_NO_CONTENT));
+  assert_true(az_iot_is_success_status(0));
+  assert_true(az_iot_is_success_status(350));
+
+  assert_false(az_iot_is_success_status(AZ_IOT_STATUS_BAD_REQUEST));
+  assert_false(az_iot_is_success_status(AZ_IOT_STATUS_TIMEOUT));
+  assert_false(az_iot_is_success_status(600));
+}
+
+static void test_az_iot_is_retriable_status_translate_success()
+{
+  assert_true(az_iot_is_retriable_status(AZ_IOT_STATUS_THROTTLED));
+  assert_true(az_iot_is_retriable_status(AZ_IOT_STATUS_SERVER_ERROR));
+
+  assert_false(az_iot_is_retriable_status(AZ_IOT_STATUS_OK));
+  assert_false(az_iot_is_retriable_status(AZ_IOT_STATUS_UNAUTHORIZED));
+}
+
+static void test_az_iot_retry_calc_delay_common_timings_success()
+{
+  assert_int_equal(2229, az_iot_retry_calc_delay(5, 1, 500, 100000, 1234));
+  assert_int_equal(321, az_iot_retry_calc_delay(5000, 1, 500, 100000, 4321));
+
+  // Operation already took more than the back-off interval.
+  assert_int_equal(0, az_iot_retry_calc_delay(10000, 1, 500, 100000, 4321));
+
+  // Max retry exceeded.
+  assert_int_equal(9995, az_iot_retry_calc_delay(5, 5, 500, 10000, 4321));
+}
+
+static void test_az_iot_retry_calc_delay_overflow_time_success()
+{
+  assert_int_equal(
+      0,
+      az_iot_retry_calc_delay(
+          INT32_MAX - 1, INT16_MAX - 1, INT32_MAX - 1, INT32_MAX - 1, INT32_MAX - 1));
+
+  assert_int_equal(
+      INT32_MAX - 1,
+      az_iot_retry_calc_delay(0, INT16_MAX - 1, INT32_MAX - 1, INT32_MAX - 1, INT32_MAX - 1));
+}
+
+#ifdef _MSC_VER
+// warning C4113: 'void (__cdecl *)()' differs in parameter lists from 'CMUnitTestFunction'
+#pragma warning(disable : 4113)
+#endif
```

Comment:
Can you please explain exactly what about `CMUnitTestFunction` is requiring this warning disable?

Reply:
It's the cmocka `void **state` parameter. I think it's not adding any value (we don't use the state feature) and it adds 3 lines of extra code.
The warning is for not using the same signature (I'm using va-args `test()` and it's expecting `test(void **)`). Using va_args (any empty () definition) is a valid C99 language feature.